okhttp-ws/src/main/java/com/squareup/okhttp/internal/ws/RealWebSocket.java

```diff
@@ -46,8 +47,9 @@
   private volatile boolean writerSentClose;
```

Comment:
This no longer requires locking. Only the application thread reads and writes to it for deciding whether to throw an `IllegalStateException`. Other threads look at the value as an indicator of whether they should try to write a close frame, but they'll ultimately catch the `IOException` if they end up missing the correct value.